brave/src/main/java/zipkin2/reporter/brave/AsyncZipkinSpanHandler.java

```diff
@@ -151,18 +156,42 @@ public Builder queuedMaxBytes(int queuedMaxBytes) {
       return (Builder) super.alwaysReportSpans(alwaysReportSpans);
     }
 
+    /**
+     * Builds an async span handler that encodes zipkin spans according to the sender's encoding.
+     */
     // AsyncZipkinSpanHandler not SpanHandler, so that Flushable and Closeable are accessible
     public AsyncZipkinSpanHandler build() {
-      return new AsyncZipkinSpanHandler(this);
+      switch (encoding) {
+        case JSON:
+          return build(new JsonV2Encoder(errorTag));
+        default:
+          throw new UnsupportedOperationException(encoding.name());
+      }
+    }
+
+    /**
+     * Builds an async span handler that encodes zipkin spans according to the encoder.
+     *
+     * <p>Note: The input encoder must use the same error tag implementation as configured by
+     * {@link #errorTag(Tag)}.
+     *
+     * @since 3.1
+     */
+    // AsyncZipkinSpanHandler not SpanHandler, so that Flushable and Closeable are accessible
+    public AsyncZipkinSpanHandler build(BytesEncoder<MutableSpan> encoder) {
```

Comment:
I noticed that we can't really move off zipkin types without this, as this handler only worked with json! Will release 3.1.0 shortly afterwards.